hazelcast/src/main/java/com/hazelcast/internal/blackbox/impl/BlackboxImpl.java

```diff
@@ -0,0 +1,321 @@
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.get(parameter);
+
+            if (sensor == null) {
+                sensor = new SensorImpl(parameter, this);
+                sensors.put(parameter, sensor);
+            }
+
+            if (sensor.view != null) {
+                return sensor.view;
+            }
+
+            SensorViewImpl view = new SensorViewImpl(this, parameter, logger);
+            sensor.view = view;
+            view.input = sensor.input;
+            return view;
+        }
+    }
+
+    public void destroy(String parameter) {
+        checkNotNull(parameter, "parameter can't be null");
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.remove(parameter);
+            if (sensor == null) {
+                return;
+            }
+
+            if (sensor.view != null) {
+                sensor.view.input = null;
+            }
+            sensor.input = null;
+            sensor.view = null;
+
+            if (logger.isFinestEnabled()) {
+                logger.finest("Destroying sensor " + parameter);
+            }
+
+            modCount.incrementAndGet();
+        }
+    }
+
+    @Override
+    public void scheduleAtFixedRate(final Runnable publisher, long delay, TimeUnit timeUnit) {
+        Runnable publishTask = new Runnable() {
+            @Override
+            public void run() {
+                publisher.run();
+            }
+        };
+        scheduledExecutorService.scheduleAtFixedRate(publishTask, 0, delay, timeUnit);
+    }
+
+    boolean containsSensor(String parameter) {
+        return sensors.containsKey(parameter);
+    }
+
+    @Override
+    public Sensor register(String parameter, final AtomicLong atomicLong) {
+        checkNotNull(atomicLong, "atomicLong can't be null");
+
+        LongSensorInput input = new LongSensorInput() {
+            @Override
+            public long get() {
+                return atomicLong.get();
+            }
+        };
+
+        return register(parameter, input);
+    }
+
+    @Override
+    public Sensor registerLongField(String parameter, final Object object, String inpput) {
+        final Field field = getLongField(parameter, object, inpput);
+
+        LongSensorInput input = new LongSensorInput() {
+            @Override
+            public long get() throws Exception {
+                return (Long) field.get(object);
+            }
+        };
+
+        return register(parameter, input);
+    }
+
+    private Field getLongField(String parameter, Object object, String fieldName) {
+        final Field field;
+        try {
+            field = object.getClass().getDeclaredField(fieldName);
+        } catch (NoSuchFieldException e) {
+            throw new IllegalArgumentException(format("Failed to register sensor with name '%s'", parameter), e);
+        }
+
+        if (!field.getType().isAssignableFrom(Long.TYPE)) {
```

Comment:
Could be a matter of taste, but `long.class` seems more intuitive to me than `Long.TYPE`.